src/js/popup/popup.js

```diff
@@ -959,6 +999,263 @@
           }
         },
       },
+      survey: {
+        init: async () => {
+          const survey = popup.panel.survey;
+          const surveyButtons = survey.select.surveyButtons();
+          const { premium } = await browser.storage.local.get("premium");
+          const tier = premium ? "premium" : "free";
+
+          // loop through all satisfaction buttons
+          surveyButtons.forEach(async (button) => {
+            // check if button is disabled
+            if (button.hasAttribute("disabled")) {
+              // do nothing and exit early
+              return;
+            }
+
+            // button is not disabled, add event listener
+            button.addEventListener("click", async (e) => {
```

Comment:
UX Suggestion: I recommend closing the panel after running this click. See `window.close()` elsewhere [in this file](https://github.com/mozilla/fx-private-relay-add-on/blob/main/src/js/popup/popup.js#L66).

Reply:
Wouldn't this stop the user from clicking on 'take the survey'?

Reply:
Yeah, not going to implement this as it would stop the user from being able to click on 'take the survey', but I did implement `window.close()` once the user clicks on 'take the survey'.